assets/js/base/components/validation/validation-input-error.js

```diff
@@ -26,7 +27,9 @@ export const ValidationInputError = ( {
 
 	return (
 		<div className="wc-block-form-input-validation-error" role="alert">
-			<p id={ getValidationErrorId( elementId ) }>{ errorMessage }</p>
+			<p id={ getValidationErrorId( elementId ) }>
+				{ decodeEntities( errorMessage ) }
```

Comment:
What would you think on decoding it in `useStoreCartCoupons`:
https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/base/hooks/cart/use-store-cart-coupons.js#L62

This way we already store the error message decoded.